### ISSUE TITLE
Bug: [no-unnecessary-template-expression] has a few bugs around comment detection

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#10684]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.tsx&code=MYewdgzgLgBBAWICuAbAJgORFAYighgOYCMMAvDAAZj5gBcMAJAN4D0AVDAJYC2ADiABOUWrHasYGfBgC%2BlAFChIsBMnRZcBQgCZyVJGDQBTAGZcwRtAxYdu-ISLBiJB42Yto5i8NDiJUmNh4RADMepTm7lxQAJ7WbJy8AsKiMOIwAJJgUbFyQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK18iALbF4QifQAexaIMocm6KAHcRG8GAC%2BIA0A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
const shouldNotFlag1 = `nan: ${/* important */ NaN}`
const shouldNotFlag2 = `undefined: ${/* important */ undefined}`
const shouldNotFlag3 = `infinity: ${/* important */ Infinity}`
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-unnecessary-template-expression": "warn"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expected none of these lines to flag

### Actual Result

These lines all flag

### Additional Info

This is just a matter of the comment check should come first: 

https://github.com/typescript-eslint/typescript-eslint/blob/b17c7f25af620c22ab9a27c270c964a6fd29de7c/packages/eslint-plugin/src/rules/no-unnecessary-template-expression.ts#L159-L203
